src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -110,47 +112,89 @@ def get_comment_contents(issue_list: List[int]) -> List[PullRequestIssue]:
     ]
 
 
-def comment_workflow():
-    pr_list = pr_to_issue_query()
-    for (gh_repo_id, pr_key, org_id, issue_list) in pr_list:
-        try:
-            organization = Organization.objects.get_from_cache(id=org_id)
-        except Organization.DoesNotExist:
-            logger.error("github.pr_comment.org_missing")
-            continue
-
-        # currently only posts new comments
-        if not features.has("organizations:pr-comment-bot", organization):
-            # or pr.summary_comment_meta is None:
-            continue
-
-        try:
-            project = Group.objects.get_from_cache(id=issue_list[0]).project
-        except Group.DoesNotExist:
-            logger.error("github.pr_comment.group_missing", extra={"organization_id": org_id})
-            continue
-
-        top_5_issues = get_top_5_issues_by_count(issue_list, project)
-        issue_comment_contents = get_comment_contents([issue["group_id"] for issue in top_5_issues])
-
-        try:
-            repo = Repository.objects.get(id=gh_repo_id)
-        except Repository.DoesNotExist:
-            logger.error("github.pr_comment.repo_missing", extra={"organization_id": org_id})
-            continue
-
-        integration = integration_service.get_integration(integration_id=repo.integration_id)
-        if not integration:
-            logger.error("github.pr_comment.integration_missing", extra={"organization_id": org_id})
-            continue
-
-        installation = integration_service.get_installation(
-            integration=integration, organization_id=org_id
+def create_or_update_comment(
+    pr_comment: PullRequestComment | None,
+    client: GitHubAppsClient,
+    repo: Repository,
+    pr_key: int,
+    comment_body: str,
+    pullrequest_id: int,
+    issue_list: List[int],
+):
+    # client will raise ApiError if the request is not successful
```

Comment:
Should we capture the error?

Reply:
i feel like we should let it be raised so we're notified through Sentry?